pkg/cloud/vsphere/services/loadbalancer/aws/loadbalancers.go

```diff
@@ -0,0 +1,315 @@
+			if err != nil {
+				return nil, nil, err
+			}
+			loadBalancerArn = createLoadBalancerOutput.LoadBalancers[0].LoadBalancerArn
+			loadBalancerDNS = createLoadBalancerOutput.LoadBalancers[0].DNSName
+
+			return loadBalancerArn, loadBalancerDNS, nil
+		}
+		return nil, nil, err
+	}
+
+	loadBalancerArn = describeLoadBalancersOutput.LoadBalancers[0].LoadBalancerArn
+	loadBalancerDNS = describeLoadBalancersOutput.LoadBalancers[0].DNSName
+
+	return loadBalancerArn, loadBalancerDNS, nil
+}
+
+func (svc *Service) reconcileListeners(loadBalancerArn *string, targetGroupArn *string) (*int64, error) {
+	var listenerPort *int64
+	describeListnerInput := &elbv2.DescribeListenersInput{
+		LoadBalancerArn: loadBalancerArn,
+	}
+	describeListnerOutput, err := svc.ELB.DescribeListeners(describeListnerInput)
+
+	if err != nil || len(describeListnerOutput.Listeners) == 0 {
+		if IsNotFound(err) || len(describeListnerOutput.Listeners) == 0 {
+			listenerInput := &elbv2.CreateListenerInput{
+				LoadBalancerArn: loadBalancerArn,
+				Port:            aws.Int64(int64(6443)),
+				Protocol:        aws.String(elbv2.ProtocolEnumTcp),
+			}
+
+			listenerInput.DefaultActions = append(listenerInput.DefaultActions, &elbv2.Action{
+				TargetGroupArn: targetGroupArn,
+				Type:           aws.String(elbv2.ActionTypeEnumForward),
+			})
+
+			listenerOutput, err := svc.ELB.CreateListener(listenerInput)
+			if err != nil {
+				return nil, err
+			}
+			return listenerOutput.Listeners[0].Port, nil
+		}
+		return nil, err
+	}
+
+	modifyListenerInput := &elbv2.ModifyListenerInput{
+		Port:        aws.Int64(int64(6443)),
+		Protocol:    aws.String(elbv2.ProtocolEnumTcp),
+		ListenerArn: describeListnerOutput.Listeners[0].ListenerArn,
+	}
+	modifyListenerInput.DefaultActions = append(modifyListenerInput.DefaultActions, &elbv2.Action{
+		TargetGroupArn: targetGroupArn,
+		Type:           aws.String(elbv2.ActionTypeEnumForward),
+	})
+	modifyListenerOutput, err := svc.ELB.ModifyListener(modifyListenerInput)
+	if err != nil {
+		return nil, err
+	}
+	listenerPort = modifyListenerOutput.Listeners[0].Port
+	return listenerPort, nil
+}
+
+func (svc *Service) deleteLoadBalancer(clusterName string) error {
+
+	describeLoadBalancersInput := &elbv2.DescribeLoadBalancersInput{
+		Names: []*string{aws.String(generateELBName(clusterName))},
+	}
+	describeLoadBalancersOutput, err := svc.ELB.DescribeLoadBalancers(describeLoadBalancersInput)
+	if err != nil {
+		if IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	loadBalancerArn := describeLoadBalancersOutput.LoadBalancers[0].LoadBalancerArn
+	deleteLoadBalancerInput := &elbv2.DeleteLoadBalancerInput{
+		LoadBalancerArn: loadBalancerArn,
+	}
+	_, err = svc.ELB.DeleteLoadBalancer(deleteLoadBalancerInput)
+	return err
+}
+
+func (svc *Service) deleteTargetGroup(clusterName string) error {
+	describeTargetGroupInput := &elbv2.DescribeTargetGroupsInput{}
```

Comment:
Hi @yastij,

Please simplify as:

```golang
describeTargetGroupInput := &elbv2.DescribeTargetGroupsInput{
	Names: []*string{aws.String(clusterName+"-controlPlane")},
}
```

Also, please note that I recommended above to extract the building of this name into a distinct function.